openedx/core/djangoapps/user_api/accounts/tests/test_permissions.py

```diff
@@ -0,0 +1,40 @@
+"""
+Tests for User deactivation API permissions
+"""
+from django.test import TestCase, RequestFactory
+
+from openedx.core.djangoapps.user_api.accounts.permissions import CanDeactivateUser
+from student.tests.factories import ContentTypeFactory, PermissionFactory, SuperuserFactory, UserFactory
+
+
+class CanDeactivateUserTest(TestCase):
+    """ Tests for user deactivation API permissions """
+
+    def setUp(self):
+        super(CanDeactivateUserTest, self).setUp()
+        self.request = RequestFactory().get('/test/url')
+
+    def test_api_permission_superuser(self):
+        self.request.user = SuperuserFactory()
+
+        result = CanDeactivateUser().has_permission(self.request, None)
+        self.assertTrue(result)
+
+    def test_api_permission_user_granted_permission(self):
+        user = UserFactory()
+        permission = PermissionFactory(
+            codename='can_deactivate_users',
+            content_type=ContentTypeFactory(
+                app_label='student'
```

Comment:
Ah! In this case the override _is_ necessary. I missed that. Yeah, what you have is fine. For completeness, however, I would `app_label` and `codename` equal to `factory.Fake('word')` in the  factories. This way the factories can be used by future developers with minimal modification if the types/codenames don't matter.

Note: This is not show-stopping. You may choose to ignore me. I will not be offended.

Info on `faker` at https://factoryboy.readthedocs.io/en/latest/fuzzy.html.
Example at https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/tests/factories.py#L170-L171.